### PR TITLE
[serve] Deflake `test_replica_startup_status_transitions ` by `await`ing signal actor in constructor

### PR DESCRIPTION
**Pull request by edoakes**
## Why are these changes needed?

`test_cluster: test_replica_startup_status_transitions` is periodically flaky with the replica hanging in `PENDING_ALLOCATION`. This could be because there is no ordering guarantee on async actor calls, so the `reconfigure` method might execute first and block the asyncio loop (due to `ray.get`), not allowing the `is_allocated` call to run.

This is a bit of a long shot, because if this was the case I'd expect it to be failing much more often than it is.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by edoakes** [comment]
A more proper fix might be to not kick off the `reconfigure` call until the `is_allocated` call is finished, but that'd be a bigger refactor for a very minute edge case.

File: python/ray/serve/tests/test_cluster.py [hunk 1]
```diff
@@ -130,8 +130,8 @@ def test_replica_startup_status_transitions(ray_cluster):
 
     @serve.deployment(version="1", ray_actor_options={"num_cpus": 2})
     class E:
-        def __init__(self):
-            ray.get(signal.wait.remote())
+        async def __init__(self):
+            await signal.wait.remote()
```
Comment (jiaodong) [inline python/ray/serve/tests/test_cluster.py:134]: i couldn't think of a case in serve that `ray.get(ref)` is more favorable compare to `await ref`, should we always use asyncio context for serve tests, and maybe even detect and recommend user to use async/await as warning message ?

**Review by jiaodong — APPROVED** [review]

**Review by iasoon — APPROVED** [review]
Nice catch! Going over the code, I think this is indeed a bug. Not that impactful, but it might cause users to get misleading warning messages when they write a blocking/slow init.

Shall I have a look at that in a separate PR?

**Comment by edoakes** [comment]
@iasoon that would be fantastic!
